server/channels/app/server.go

```diff
@@ -138,19 +135,15 @@ type Server struct {
 
 	Audit *audit.Audit
 
-	joinCluster bool
-	// startSearchEngine bool
+	joinCluster  bool
 	skipPostInit bool
 
 	Cloud       einterfaces.CloudInterface
 	IPFiltering einterfaces.IPFilteringInterface
 
 	tracer *tracing.Tracer
 
-	skipProductsInit bool
-
-	products map[string]product.Product
-	services map[product.ServiceKey]any
+	ch *Channels
```

Comment:
This is the most bothersome part. The hierarchy is Server->Channels->App, so why does Server need to know Channels? Anyway, it's not now, `Server` did hold `Channels` before via `products`.

Reply:
Actually the idea was to separate App and Server so that we could separate housekeeping and managing (server) work from the Application. In fact run the server w/o any Application code (eg for a standalone Boards app). And let App own Channels, Boards as products etc.